core/trino-main/src/main/java/io/trino/sql/planner/ExpressionInterpreter.java

```diff
@@ -969,6 +972,26 @@ protected Object visitNotExpression(NotExpression node, Object context)
         @Override
         protected Object visitLogicalExpression(LogicalExpression node, Object context)
         {
+            // Try simplifying expressions using TupleDomain, e.g. a > 1 and a > 2 => a > 2
```

Comment:
Would it make sense to extract this new part into a merhod like 
`Optional<Object> trySimplifyExpressionToTupleDomain(...)`?